app/controllers/application_pets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class ApplicationPetsController < ApplicationController
+  def update
+    application = Application.find(params[:id])
```

Comment:
This is very minor, but from a routing perspective, I would assume the ID in `/application_pets/:id` would be the primary key for that ApplicationPet record rather than the application ID. So, an equally RESTful route with the IDs of the separate resources (applications and pets) could be: `"/applications/:application_id/pets/:pet_id"`

You can also keep "/application_pets/:id"  as it is RESTful, but it would be more intuitive to pass the joins record ID here (but it is not as easily readable from your view, hence why it's a bit tricker for this context)